From: Arnaud Ferraris <aferraris@debian.org>
Date: Thu, 29 Aug 2024 19:13:03 +0200
Subject: arm64: dts: rk3399-pinephone-pro: Add individual LED function

Currently the LED function is only defined for the `multi-led` group.
However, this doesn't trickle down to individual LEDs being part of this
group, so those are named `<color>:` instead of `<color>:indicator`. As
a consequence, they're not recognized by `feedbackd` which then picks
the flashlight as the only available LED.

Adding a `function` attribute to individual LED nodes restores the
naming `feedbackd` expects and fixes this issue.

Signed-off-by: Arnaud Ferraris <aferraris@debian.org>
---
 arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
index 32cffc3..cc1e70f 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
@@ -246,16 +246,19 @@ leds {
 
 		led_red: led-0 {
 			color = <LED_COLOR_ID_RED>;
+			function = LED_FUNCTION_INDICATOR;
 			gpios = <&gpio4 RK_PD2 GPIO_ACTIVE_HIGH>;
 		};
 
 		led_green: led-1 {
 			color = <LED_COLOR_ID_GREEN>;
+			function = LED_FUNCTION_INDICATOR;
 			gpios = <&gpio4 RK_PD5 GPIO_ACTIVE_HIGH>;
 		};
 
 		led_blue: led-2 {
 			color = <LED_COLOR_ID_BLUE>;
+			function = LED_FUNCTION_INDICATOR;
 			gpios = <&gpio4 RK_PD6 GPIO_ACTIVE_HIGH>;
 		};
 	};
